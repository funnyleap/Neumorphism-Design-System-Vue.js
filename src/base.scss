html,
body {
  height: 100%;
}

*,
*::before,
*::after {
  box-sizing: border-box;
}

body {
  min-width: 768px;
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-family: 'Roboto', Arial, Helvetica, sans-serif;
  color: var(--color-dark-silver);
  background-color: var(--color-back-gray);
}

.wrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  height: 100%;
  overflow-x: hidden;

  &__content {
    display: block;
    flex-grow: 1;
    flex-shrink: 0;
    flex-basis: auto;
    width: 100%;
  }

  &_demo {
    .notifications {
      margin-top: 40px;
    }
  }
  @media (max-width: 1024px) {
    width: auto;
  }
}

@media (max-width: 768px) {
  .container {
    max-width: 720px;
  }
}

.input-required-wrapper {
  position: relative;
}

.page-wrap {
  .page-wrap__top {
    padding-top: var(--page-wrap-padding);
    padding-bottom: calc(var(--page-wrap-padding) * 1.5);
  }

  .page-wrap__top-prev {
    font-weight: 600;
    font-size: 18px;
    line-height: 27px;
    color: var(--color-dark-silver);
  }

  .page-wrap__top-btn {
    a {
      display: block;
      color: inherit;
    }
  }

  .page-wrap__title {
    margin-top: 0;
    margin-bottom: 0;
    font-weight: 500;
  }

  .page-wrap__content {
    padding-bottom: var(--page-wrap-padding);
  }

  .page-wrap__tabs {
    margin-bottom: 30px;
  }
}

.tab-list {
  position: relative;

  .tabs__items {
    width: 100%;

    .tabs__item {
      justify-content: center;
      width: 272px;
      text-align: center;

      .breath-r {
        margin-right: 0;
        padding: 10px 16px;
      }
    }
  }
}

.impers-modal {
  width: 470px;
  padding: 20px;
  font-size: 12px;
  text-align: center;

  &__header {
    margin: 0;
  }

  &__type {
    margin: 38px 0 0 0;
    font-weight: 700;
    color: var(--warm-grey-color);
  }

  &__email {
    margin: 0 0 20px 0;
    color: var(--warm-grey-color);
    border-bottom: 1px solid var(--color-pale-gray);
  }

  &__actions {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 38px 0 0 0;
  }
}
// imperson end

.custom-modal {
  width: 440px;
  margin: 16px;

  .custom-modal__info {
    margin-top: 38px;
  }

  .custom-modal__id {
    font-weight: 500;
    font-size: 12px;
    color: var(--warm-grey-color);
  }

  .custom-modal__id-value {
    margin-top: 2px;
    padding-bottom: 2px;
    font-size: 12px;
    line-height: 16px;
    color: var(--warm-grey-color);
    border-bottom: 1px solid var(--light-grey);
  }

  .custom-modal__desc {
    font-size: 12px;
    line-height: 16px;
  }

  .custom-modal__buttons {
    margin-top: 50px;
  }

  h2 {
    margin: 0;
    font-size: 12px;
    line-height: 12px;
  }

  a {
    color: var(--header-bg-color);
  }
}

.datepicker-block {
  position: relative;
}

.input-star {
  position: absolute;
  top: -19px;
  left: -19px;
}

.notifications {
  max-width: 632px;
  margin-top: 0;
}

body .notify {
  position: relative;

  display: flex;
  justify-content: flex-start;
  align-items: center;
  min-height: var(--header-height);
  margin: 0 0 5px;
  padding: 16px 56px 16px 96px;

  font-size: 14px;
  line-height: 24px;

  color: var(--color-white);
  background-color: var(--color-dark-blue);
  border-left: 5px solid var(--color-dark-blue);

  &::before {
    content: '';
    position: absolute;
    top: 50%;
    left: 24px;
    width: 32px;
    height: 32px;
    background-image: url('#{$--icon-path}/demo-icons/demo-warning.svg');
    background-repeat: no-repeat;
    transform: translateY(-50%);
  }

  .notification-content {
    display: block;
  }

  a {
    color: var(--color-white);
    text-decoration: underline;
  }

  &.warn {
    background-color: var(--color-dull-yellow);
    border-left-color: var(--color-dull-yellow);
  }

  &.error {
    background-color: var(--color-reddish);
    border-left-color: var(--color-reddish);

    &::before {
      background-image: url('#{$--icon-path}/demo-icons/demo-error.svg');
    }
  }

  &.success {
    background-color: var(--color-greenish);
    border-left-color: var(--color-greenish);

    &::before {
      background-image: url('#{$--icon-path}/demo-icons/demo-success.svg');
    }
  }
}

button {
  cursor: pointer;
}

.typeahead {
  .form-control {
    height: 32px;
    font-size: 14px;
    color: var(--color-black);
    letter-spacing: 0.2px;

    &:focus {
      border-color: var(--color-water-blue);
      box-shadow: none;
    }
  }

  .list-group {
    &.shadow {
      box-shadow: none !important;
    }
  }

  .list-group-item {
    padding: 6px 12px;
    font-size: 14px;
    color: var(--color-dark-gray);
    letter-spacing: 0.2px;
    background-color: var(--color-white);
    border: none;
    border-bottom: 1px solid var(--color-white);
    cursor: pointer;

    strong {
      font-weight: 400;
      color: var(--color-black);
    }

    &:hover {
      background-color: var(--color-white);
    }
  }
}

input[type='number']::-webkit-inner-spin-button,
input[type='number']::-webkit-outer-spin-button {
  appearance: none;
  margin: 0;
}

@-moz-document url-prefix() {
  input[type='number'] {
    appearance: textfield;
  }
}

.field-message-error {
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 4px;
  font-size: 12px;
  line-height: 16px;
  color: var(--color-reddish);
  letter-spacing: 0.2px;
}

body .text-error {
  color: var(--color-reddish);
}
