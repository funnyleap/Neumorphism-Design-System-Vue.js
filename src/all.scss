@import './fonts/index.scss';
@import './main.scss';

@import './neumorphism/index.scss';

@import './icons/index.scss';
