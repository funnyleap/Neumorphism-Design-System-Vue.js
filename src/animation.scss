.bounce-enter-active {
  animation: bounce-in 0.5s;
}

.bounce-leave-active {
  animation: bounce-in 0.5s reverse;
}

.slide-fade-enter-active {
  opacity: 0;
  transition: all 0.1s ease-in;
}

.slide-fade-leave-active {
  transition: all 0.1s ease-in;
}

.slide-fade-enter,
.slide-fade-leave-to {
  transform: translateX(10px);
  opacity: 0;
}

.slide-fade-down-enter-active {
  opacity: 0;
  transition: all 0.1s ease-in;
}

.slide-fade-down-leave-active {
  transition: all 0.1s ease-in;
}

.slide-fade-down-enter,
.slide-fade-down-leave-to {
  transform: translateY(-10px);
  opacity: 0;
}

@keyframes bounce-in {
  0% {
    transform: scale(0);
  }

  50% {
    transform: scale(1.1);
  }

  100% {
    transform: scale(1);
  }
}

@keyframes swing {
  15% {
    transform: translateX(5px);
  }

  30% {
    transform: translateX(-5px);
  }

  50% {
    transform: translateX(3px);
  }

  65% {
    transform: translateX(-3px);
  }

  80% {
    transform: translateX(2px);
  }

  100% {
    transform: translateX(0);
  }
}

