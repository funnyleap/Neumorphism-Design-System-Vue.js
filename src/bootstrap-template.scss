.modal-content {
  border: none;
  box-shadow: none;
}