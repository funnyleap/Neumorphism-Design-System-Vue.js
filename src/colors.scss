:root {
  --color-primary: #de4b7a;
}
