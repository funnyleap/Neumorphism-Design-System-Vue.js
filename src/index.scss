@import './animation.scss';
@import './base.scss';
@import './bootstrap-template.scss';
@import './colors.scss';
@import './custom-tooltip.scss';
@import './helpers.scss';
@import './normilize.scss';
@import './typography.scss';
@import './vars.scss';