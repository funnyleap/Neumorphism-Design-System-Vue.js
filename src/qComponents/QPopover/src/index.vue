<template>
  <component :is="tagName">
    <transition
      :name="transition"
      @after-leave="destroy"
    >
      <div
        v-show="isPopoverShown"
        ref="qPopper"
        class="popover-content"
        :class="popoverContentClasses"
        :style="popoverContentStyles"
      >
        <div
          v-if="icon"
          class="popover-content__icon"
          :class="icon"
          :style="popoverIconStyles"
        />
        <div
          v-if="title"
          class="popover-content__title"
          :class="popoverTitleClasses"
        >
          {{ title }}
        </div>
        <div class="popover-content__inner">
          <slot />
        </div>
      </div>
    </transition>

    <slot name="reference" />
  </component>
</template>

<script>
import Popper from 'popper.js';

function eventOn(element, event, handler) {
  if (element && event && handler) {
    element.addEventListener(event, handler, false);
  }
}

function eventOff(element, event, handler) {
  if (element && event) {
    element.removeEventListener(event, handler, false);
  }
}

const TRIGGERS = ['click', 'hover'];

const PLACEMENTS = [
  'top',
  'top-start',
  'top-end',
  'right',
  'right-start',
  'right-end',
  'bottom',
  'bottom-start',
  'bottom-end',
  'left',
  'left-start',
  'left-end'
];

export default {
  name: 'QPopover',

  props: {
    title: {
      type: String,
      default: ''
    },
    tagName: {
      type: String,
      default: 'span'
    },
    trigger: {
      type: String,
      default: 'click',
      validator: value => TRIGGERS.includes(value)
    },
    placement: {
      type: String,
      default: 'top-start',
      validator: value => PLACEMENTS.includes(value)
    },
    icon: {
      type: String,
      default: ''
    },
    iconColor: {
      type: String,
      default: 'var(--gradient-secondary)'
    },
    transition: {
      type: String,
      default: 'fade-in-linear'
    },
    openDelay: {
      type: Number,
      default: 10
    },
    closeDelay: {
      type: Number,
      default: 10
    },
    minWidth: {
      type: [String, Number],
      default: ''
    },
    maxWidth: {
      type: [String, Number],
      default: ''
    },
    popperOptions: {
      type: Object,
      default: () => ({})
    }
  },

  data() {
    return {
      popperJS: null,
      referenceElement: null,
      isPopoverShown: false
    };
  },

  computed: {
    popoverContentClasses() {
      return {
        'popover-content_without-icon': !this.icon
      };
    },

    popoverContentStyles() {
      return {
        minWidth:
          typeof this.minWidth === 'number'
            ? `${this.minWidth}px`
            : this.minWidth,
        maxWidth:
          typeof this.maxWidth === 'number'
            ? `${this.maxWidth}px`
            : this.maxWidth
      };
    },

    popoverIconStyles() {
      const backgroundProperty = this.iconColor.includes('-gradient')
        ? 'backgroundImage'
        : 'backgroundColor';

      return {
        [backgroundProperty]: this.iconColor
      };
    },

    popoverTitleClasses() {
      return {
        'popover-content__title_no-content': !this.$slots.default
      };
    }
  },

  watch: {
    isPopoverShown(value) {
      if (value) {
        this.$emit('show');

        if (this.popperJS) {
          this.popperJS.enableEventListeners();
          this.popperJS.scheduleUpdate();
        } else {
          this.createPopper();
        }
      } else {
        if (this.popperJS) {
          this.popperJS.disableEventListeners();
        }

        this.$emit('hide');
      }
    }
  },

  created() {
    if (!this.$slots.reference && process.env.NODE_ENV !== 'production') {
      console.error('QPopover requires reference be provided in the slot.');
    }
  },

  mounted() {
    this.referenceElement = this.$slots?.reference?.[0]?.elm;

    if (!this.referenceElement) return;

    switch (this.trigger) {
      default:
      case 'hover':
        eventOn(this.referenceElement, 'mouseover', this.handleMouseOver);
        eventOn(this.referenceElement, 'mouseout', this.onMouseOut);
        eventOn(this.$refs.qPopper, 'mouseover', this.handleMouseOver);
        eventOn(this.$refs.qPopper, 'mouseout', this.onMouseOut);
        break;

      case 'click':
        eventOn(this.referenceElement, 'click', this.togglePopover);
        eventOn(document, 'click', this.handleDocumentClick);
        break;
    }
  },

  beforeDestroy() {
    eventOff(this.referenceElement, 'click', this.togglePopover);
    eventOff(this.referenceElement, 'mouseout', this.onMouseOut);
    eventOff(this.referenceElement, 'mouseover', this.handleMouseOver);
    eventOff(document, 'click', this.handleDocumentClick);

    this.isPopoverShown = false;
    this.destroy();
  },

  methods: {
    async createPopper() {
      await this.$nextTick();

      const options = {
        placement: this.placement,
        computeStyle: {
          boundariesElement: 'body',
          gpuAcceleration: false
        },
        modifiers: {
          offset: {
            offset: '0, 16px'
          }
        },
        ...this.popperOptions
      };

      this.popperJS = new Popper(
        this.referenceElement,
        this.$refs.qPopper,
        options
      );
    },

    togglePopover() {
      this.isPopoverShown = !this.isPopoverShown;
    },

    destroy() {
      this.isPopoverShown = false;

      if (this.popperJS) {
        this.popperJS.destroy();
        this.popperJS = null;
      }
    },

    handleMouseOver() {
      clearTimeout(this.timer);

      this.timer = setTimeout(() => {
        this.isPopoverShown = true;
      }, this.openDelay);
    },

    onMouseOut() {
      clearTimeout(this.timer);

      this.timer = setTimeout(() => {
        this.isPopoverShown = false;
      }, this.closeDelay);
    },

    handleDocumentClick({ target }) {
      if (
        this.$el.contains(target) ||
        this.referenceElement.contains(target) ||
        this.$refs.qPopper.contains(target)
      ) {
        return;
      }

      this.isPopoverShown = false;
    }
  }
};
</script>
