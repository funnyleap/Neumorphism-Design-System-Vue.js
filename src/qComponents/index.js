import vClickOutside from 'v-click-outside';

import QButton from './QButton';
import { QCheckbox, QCheckboxGroup } from './QCheckbox';
import { QSelect, QOption } from './QSelect';
import QInput from './QInput';
import QTextarea from './QTextarea';
import QTag from './QTag';
import QCollapse from './QCollapse';
import QCollapseItem from './QCollapseItem';
import QContextMenu from './QContextMenu';
import QPopover from './QPopover';
import QScrollbar from './QScrollbar';
import QTabPane from './QTabPane';
import QTabs from './QTabs';

const components = [
  QButton,
  QCheckbox,
  QCheckboxGroup,
  QCollapse,
  QCollapseItem,
  QContextMenu,
  QInput,
  QPopover,
  QScrollbar,
  QSelect,
  QOption,
  QTabPane,
  QTextarea,
  QTabs,
  QTag
];

const install = function(Vue) {
  Vue.use(vClickOutside);

  components.forEach(component => {
    Vue.component(component.name, component);
  });

  //   Vue.use(InfiniteScroll);
  //   Vue.use(Loading.directive);

  //   Vue.prototype.$ELEMENT = {
  //     size: opts.size || '',
  //     zIndex: opts.zIndex || 2000
  //   };

  //   Vue.prototype.$loading = Loading.service;
  //   Vue.prototype.$msgbox = MessageBox;
  //   Vue.prototype.$alert = MessageBox.alert;
  //   Vue.prototype.$confirm = MessageBox.confirm;
  //   Vue.prototype.$prompt = MessageBox.prompt;
  //   Vue.prototype.$notify = Notification;
  //   Vue.prototype.$message = Message;
};

/* istanbul ignore if */
if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue);
}

export default {
  version: '0.0.1',
  //   locale: locale.use,
  install
};
