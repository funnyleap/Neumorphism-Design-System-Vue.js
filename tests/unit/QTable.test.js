import { shallowMount } from '@vue/test-utils';
import { QTable } from '../../src/qComponents';

describe('QTable.vue', () => {
  const options = {
    propsData: {
      columns: [
        {
          key: 'col1',
          value: 'Column 1'
        }
      ],
      rows: [
        {
          col1: 'Columns row 1'
        }
      ]
    }
  };

  it('QTable should match snapshot', () => {
    const { element } = shallowMount(QTable, options);

    expect(element).toMatchSnapshot();
  });
});
