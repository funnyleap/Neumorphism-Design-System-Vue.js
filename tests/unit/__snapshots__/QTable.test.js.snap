// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QTable.vue QTable should match snapshot 1`] = `
<div
  class="q-table"
>
  <!---->
   
  <!---->
   
  <div
    class="q-table__loading-wrapper"
  >
    <q-scrollbar-stub
      theme="secondary"
      viewclass=""
      viewtag="div"
      wrapclass="q-table__scroll-wrapper"
    >
      <div
        class="q-table__wrapper"
        style="transition: all 0.2s;"
      >
        <!---->
         
        <!---->
         
        <table
          cellpadding="0"
          cellspacing="0"
          class="q-table__table"
        >
          <thead>
            <tr>
              <!---->
               
              <th
                class="q-table__header-cell  "
              >
                <div
                  class="q-table__header-cell-wrapper"
                >
                  <!---->
                   
                  <div
                    class="q-table__header-cell-content"
                  >
                    <!---->
                     
                      Column 1
                     
                    <!---->
                  </div>
                </div>
              </th>
            </tr>
          </thead>
           
          <tbody>
            <!---->
             
            <row-hoc-stub
              checked-rows=""
              checkedrows=""
              children-key="children"
              childrenkey="children"
              class=""
              columns="[object Object]"
              fixed-column=""
              fixed-column-width="200"
              fixedcolumn=""
              fixedcolumnwidth="200"
              indent-size="16"
              indentsize="16"
              row="[object Object]"
              row-index="0"
              rowindex="0"
            />
          </tbody>
        </table>
      </div>
    </q-scrollbar-stub>
  </div>
</div>
`;
